src/ray/raylet/node_manager.cc

```diff
@@ -2967,6 +2990,10 @@ MemoryUsageRefreshCallback NodeManager::CreateMemoryUsageRefreshCallback() {
           RAY_LOG(INFO) << "Latest 10 worker details:\n"
                         << this->WorkersDebugString(workers, max_to_print);
 
+          std::unique_ptr<rpc::RayException> task_failure_reason = std::make_unique<rpc::RayException>();
+          task_failure_reason->set_formatted_exception_string("out of memory!");
+          task_failure_reason->set_error_type(rpc::ErrorType::OUT_OF_MEMORY);
+          SetTaskFailureReason(latest_worker->GetAssignedTaskId(), std::move(task_failure_reason));
```

Comment:
Looks like we only take care of OOM For this now? are you planning to handle other error types? (like worker failure)

Reply:
Can we do that in a follow up PR or has to be here? Just worried it will take time away to ensure it is done properly

Reply:
confirmed we will do this in follow up